.github/workflows/ci-workflow.yml

```diff
@@ -1,58 +1,49 @@
-name: E4E Pipeline
-
-on: [push, pull_request]
-
+name: E4E CI Workflow
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - "*"
+env:
+  PROJECT_NAME: explorer-for-endevor
+  AUDIT_SEVERITY_HIGH: 7
 jobs:
-  build:
+  build-test-artifact:
     runs-on: ${{ matrix.os }}
-
-    env:
-      PROJECT_NAME: explorer-for-endevor
-
     strategy:
       # Continue to run tests on the other systems if one fails
       fail-fast: false
       matrix:
         node-version: [12.x, 14.x, 16.x]
         os: [windows-latest, ubuntu-latest, macos-latest]
-
     steps:
       - name: Check out code
         uses: actions/checkout@v2
-
       - name: Use Node.js ${{ matrix.node-version }}
         uses: actions/setup-node@v1
         with:
           node-version: ${{ matrix.node-version }}
-
       - name: Install dependencies
         run: yarn install --frozen-lockfile
-
       - name: Audit
+        run: yarn audit
         id: audit
         continue-on-error: true
-        run: yarn audit
-
       - name: Lint
+        if: steps.audit.outputs.exit_status < ${{ env.AUDIT_SEVERITY_HIGH }}
         run: yarn lint
-
       - name: Run tests
         uses: GabrielBB/xvfb-action@v1.0
+        id: tests
         with:
           run: yarn test
-
```

Comment:
Hmm, well, because it is better and cheaper to see a mistake within documentation or licensing earlier (in a PR or in a feature branch), than during publishing.